src/components/modals/ButtonsModal.tsx

```diff
@@ -62,42 +61,39 @@ export function ButtonsModal<Buttons extends { [key: string]: ButtonInfo }>(prop
   const handleCancel = useHandler(() => bridge.resolve(undefined))
 
   const containerStyle: ViewStyle = {
-    flex: fullScreen ? 1 : 0
+    flexGrow: fullScreen ? 1 : 0,
+    flexShrink: fullScreen ? 0 : 1
   }
 
+  const buttonInfo = Object.keys(buttons).map((key, i, arr) => {
+    const { label, onPress } = buttons[key]
+
+    const handlePress = (): Promise<void> | undefined => {
+      if (onPress == null) {
+        bridge.resolve(key)
+        return
+      }
+      return onPress().then(
+        result => {
+          if (result) bridge.resolve(key)
+        },
+        error => showError(error)
+      )
+    }
+
+    return { label, onPress: handlePress }
+  })
+
   return (
     <ModalUi4 warning={warning} bridge={bridge} title={title} onCancel={disableCancel ? undefined : handleCancel}>
       <View style={[styles.textStyle, containerStyle]}>
         {message != null ? <Paragraph>{message}</Paragraph> : null}
         {children}
-      </View>
-      <View style={styles.buttonsStyle}>
-        <View style={styles.innerButtonStyle}>
-          {Object.keys(buttons).map((key, i, arr) => {
-            let defaultType: 'primary' | 'secondary'
-            if (theme.preferPrimaryButton) {
-              defaultType = i === 0 ? 'primary' : 'secondary'
-            } else {
-              defaultType = i === 0 && arr.length > 1 ? 'primary' : 'secondary'
-            }
-            const { type = defaultType, label, onPress } = buttons[key]
-
-            const handlePress = (): Promise<void> | undefined => {
-              if (onPress == null) {
-                bridge.resolve(key)
-                return
-              }
-              return onPress().then(
-                result => {
-                  if (result) bridge.resolve(key)
-                },
-                error => showError(error)
-              )
-            }
-
-            return <ButtonUi4 key={key} label={label} marginRem={0.5} type={type} onPress={handlePress} layout="column" />
-          })}
-        </View>
+        <ModalButtons
+          primary={buttonInfo.length > 0 ? buttonInfo[0] : undefined}
+          secondary={buttonInfo.length > 1 ? buttonInfo[1] : undefined}
+          tertiary={buttonInfo.length > 2 ? buttonInfo[2] : undefined}
+        />
```

Comment:
This is a pretty big change! All the buttons modals in the app are going to have their buttons types affected, which will be a big QA effort. This should be its own PR, and in that PR, we'll have to audit all the places where we use this, and we will have to remove the `theme.preferPrimaryButton`.

Reply:
Yeah this bigger change came from when I resolved with Madison that this is the way we are going regarding modal buttons. I confirmed with Paul if it was safe for white labels to deprecate `theme.preferPrimaryButton` and this is the last remaining place we use `theme.preferPrimaryButton`.

Reply:
Created https://github.com/EdgeApp/edge-react-gui/pull/5009